src/Interpreters/MutationsInterpreter.cpp

```diff
@@ -1365,6 +1359,21 @@ void MutationsInterpreter::validate()
         }
     }
 
+    const auto & storage_columns = source.getStorageSnapshot(metadata_snapshot, context)->metadata->getColumns();
+    for (const auto & command : commands)
+    {
+        for (const auto & [column_name, _] : command.column_to_update_expression)
+        {
+            auto column = storage_columns.tryGetColumn(GetColumnsOptions::Ordinary, column_name);
+            if (column && column->type->hasDynamicSubcolumns())
```

Comment:
Just moved this check from `MutationsInterpreter` constructor to `MutationsInterpreter::validate`. This exception from the constructor didn't processed correctly in a very specific case found by fuzzer